src/library.js

```diff
@@ -445,9 +445,6 @@ LibraryManager.library = {
   // TODO: There are currently two abort() functions that get imported to asm module scope: the built-in runtime function abort(),
   // and this function _abort(). Remove one of these, importing two functions for the same purpose is wasteful.
   abort__sig: 'v',
-  // Proxy synchronously, which will have the effect of halting the program
-  // and killing all threads, including this one.
-  abort__proxy: 'sync',
   abort: function() {
```

Comment:
Please add a comment here, maybe
```cpp
// abort() must be proxied, but we avoid the normal method since
// the runtime may not be in a good enough state to support that.
// Instead, it is proxied using postMessage, see library_pthread.js.

Reply:
Its not that abort needs to be proxies its just the `onAbort` callback.  I will add a comment.

Reply:
> (Is there an existing test for this?)

Yes, the test I added back in #14669 tests exactly this.    It injects the `onAbort` handler during preInit which means that handler will only exist on the main thread and therefore must be invoked there: 

https://github.com/emscripten-core/emscripten/blob/8cbca51742af5085ef7003fa95844a4f27a95305/tests/test_core.py#L2356-L2360

Reply:
Sounds good.